src/DGtal/images/FunctorConstImage.h

```diff
@@ -0,0 +1,341 @@
+
+      inline ConstReverseIterator rbegin()  const { return ConstReverseIterator( end() ); }
+      inline ConstReverseIterator rbegin( Point const& aPoint ) const { return ConstReverseIterator( ++begin(aPoint) ); }
+      inline ConstReverseIterator rend()    const { return ConstReverseIterator( begin() ); }
+
+  private:
+      Self const& myFunctorConstImage;
+    }; // End of class ConstRange
+}; // End of class FunctorConstImage
+
+/** @brief Overloads 'operator<<' for displaying objects of class @ref FunctorConstImage.
+ *
+ * @param out the output stream where the object is written.
+ * @param object the object of class @ref FunctorConstImage to write.
+ * @return the output stream after the writing.
+ */
+template <typename TDomain, typename TValue, typename TFunctor>
+std::ostream&
+operator<< ( std::ostream & out, const FunctorConstImage<TDomain, TValue, TFunctor> & object )
+{
+  object.selfDisplay(out);
+  return out;
+}
+
+/** @brief FunctorConstImage construction helper with specification of the return type.
+ *
+ * @tparam  TValue    The image value type.
+ * @tparam  TDomain   The domain type (auto-deduced).
+ * @tparam  TFunctor  The functor type (auto-deduced).
+ * @param   aDomain   The image domain.
+ * @param   aFunctor  The functor that generates the image.
+ * @return an instance of the appropriate FunctorConstImage type.
+ *
+ * @see FunctorConstImage
+ */
+template <
+  typename TValue,
+  typename TDomain,
+  typename TFunctor
+>
+inline auto
+makeFunctorConstImage( TDomain const& aDomain, TFunctor && aFunctor )
+    -> FunctorConstImage<TDomain, TValue, decltype(holdFunctor(std::forward<TFunctor>(aFunctor)))>
+  {
+    return FunctorConstImage<TDomain, TValue, decltype(holdFunctor(std::forward<TFunctor>(aFunctor)))>{ aDomain, holdFunctor(std::forward<TFunctor>(aFunctor)) };
+  }
+
+//@{
+
+/** @brief FunctorConstImage construction helper with auto-deduction of the return type.
+ *
+ * @tparam  TDomain   The domain type (auto-deduced).
+ * @tparam  TFunctor  The functor type (auto-deduced).
+ * @param   aDomain   The image domain.
+ * @param   aFunctor  The functor that generates the image.
+ * @return an instance of the appropriate FunctorConstImage type.
+ *
+ * @see FunctorConstImage and @ref moduleImages for more informations.
+ */
+template <
+  typename TDomain,
+  typename TFunctor
+>
+inline auto
+makeFunctorConstImage( TDomain const& aDomain, TFunctor && aFunctor )
+    -> FunctorConstImage<
+          TDomain,
+          typename std::decay<decltype(aFunctor(std::declval<typename TDomain::Point>()))>::type,
+          decltype(holdFunctor(std::forward<TFunctor>(aFunctor)))
+        >
+  {
+    return FunctorConstImage<
+        TDomain,
+        typename std::decay<decltype(aFunctor(std::declval<typename TDomain::Point>()))>::type,
+        decltype(holdFunctor(std::forward<TFunctor>(aFunctor)))
+      >{ aDomain, holdFunctor(std::forward<TFunctor>(aFunctor)) };
+  }
+
```

Comment:
Is doxygen able to group these two functions for doc ?

Reply:
It works at least with methods but here, I only see one entry for this function ... I will duplicate the doc :wink: